### PR TITLE
Fix logarchive collection on Xcode 8.3

### PR DESCRIPTION
**Pull request by endocrimes**
This changes log archive collection to specify the `--output` parameter on
`simctl spawn {id} log collect` to fix an issue where the log archive would be
written to an incorrect path when running undner Xcode 8.3

- Does this parameter exist on 8.x < 8.3? (if this PR passes, then it works at least on 8.2)

8.3 PR Failing because of this here:
- https://github.com/fastlane/fastlane/pull/8829

Example project:
- https://circleci.com/gh/DanToml/Lock.swift/tree/dani_8.3_fl

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Scan Scan::TestCommandGenerator with Scan option :include_simulator_logs extract system.logarchive copies all device logs to the output directory | <a href='https://github.com/fastlane/fastlane/blob/685f545946b1ecbd68208d17909b7e957fb8e2dc/./scan/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator copy_simulator_logs copies all device log archives to the output directory on macOS 10.12 (Siera) | <a href='https://github.com/fastlane/fastlane/blob/685f545946b1ecbd68208d17909b7e957fb8e2dc/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by DanToml** [comment]
Ok looks like this behaviour is actually different in 8.0..<8.2 and 8.3

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Scan Scan::TestCommandGenerator with Scan option :include_simulator_logs extract system.logarchive copies all device logs to the output directory | <a href='https://github.com/fastlane/fastlane/blob/7e7de74eefa74d47a8c72d1039d4fa5635ab4612/./scan/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator copy_simulator_logs copies all device log archives to the output directory on macOS 10.12 (Siera) | <a href='https://github.com/fastlane/fastlane/blob/7e7de74eefa74d47a8c72d1039d4fa5635ab4612/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Scan Scan::TestCommandGenerator with Scan option :include_simulator_logs extract system.logarchive copies all device logs to the output directory | <a href='https://github.com/fastlane/fastlane/blob/ed4333acf3a654f5d727215dd9335949654a5d72/./scan/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator copy_simulator_logs copies all device log archives to the output directory on macOS 10.12 (Siera) | <a href='https://github.com/fastlane/fastlane/blob/ed4333acf3a654f5d727215dd9335949654a5d72/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Snapshot Snapshot::TestCommandGenerator copy_simulator_logs copies all device log archives to the output directory on macOS 10.12 (Siera) | <a href='https://github.com/fastlane/fastlane/blob/ae26e2dbd171552e3260d84169f2c29e98df5eb0/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: fastlane_core/lib/fastlane_core/device_manager.rb [hunk 1]
```diff
@@ -250,20 +250,12 @@ def copy_logfile(device, log_identity, logs_destination_dir)
...
+        logarchive_dst = Shellwords.escape(File.join(logs_destination_dir, "system_logs-#{log_identity}.logarchive"))
         FileUtils.rm_rf(logarchive_dst)
-
-        command = "xcrun simctl spawn #{device.udid} log collect 2>/dev/null"
+        command = "xcrun simctl spawn #{device.udid} log collect --output #{logarchive_dst} 2>/dev/null"
         FastlaneCore::CommandExecutor.execute(command: command, print_all: false, print_command: true)
-
-        FileUtils.mkdir_p(logarchive_dst)
-        FileUtils.cp_r("#{logarchive_src}/.", logarchive_dst)
-        UI.success "Copying file '#{logarchive_src}' to '#{logarchive_dst}'..."
```
Comment (lyndsey-ferguson) [inline fastlane_core/lib/fastlane_core/device_manager.rb]: I like the log message, can we have it back?

**Review by lyndsey-ferguson — APPROVED** [review]
Other than my ask, looks good (and making sure it works with Xcode 8.2.1)

File: fastlane_core/lib/fastlane_core/device_manager.rb (hunk 1 above)
Comment (DanToml) [inline fastlane_core/lib/fastlane_core/device_manager.rb]: @lyndsey-ferguson We never actually copy from a src to a destination dir now tho? - We could potentially say something like "Exporting logarchive to {path}"?

**Comment by DanToml** [comment]
image
Appears to have the same option on 8.2.1 :D

File: fastlane_core/lib/fastlane_core/device_manager.rb (hunk 1 above)
Comment (lyndsey-ferguson) [inline fastlane_core/lib/fastlane_core/device_manager.rb]: I had thought of using the return value from the `execute`, but looking at this, we are already printing the command, so I don't think this is necessary.

Comment (DanToml) [inline fastlane_core/lib/fastlane_core/device_manager.rb]: :+1:

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
Beautiful, thanks @DanToml 👍

**Pull request merged**